arch/TriCore/TriCoreInstPrinter.c

```diff
@@ -269,20 +267,24 @@ static void printDisp24Imm(MCInst *MI, int OpNum, SStream *O)
 		switch (MCInst_getOpcode(MI)) {
 		case TRICORE_CALL_b:
 		case TRICORE_FCALL_b: {
-			res = DISP1(24);
+			res = DISP_SEXT_2ALIGN(24);
 			break;
 		}
 		case TRICORE_CALLA_b:
 		case TRICORE_FCALLA_b:
+			// disp24[23:20], 7'b0, disp24[19:0], 1'b0}
+			res = ((disp & 0xf00000ULL) << 2) |
+-			      ((disp & 0xfffffULL) << 1);
```

Comment:
I can probably see what you mean elsewhere, why did you change it here?
It looks like this matches the ISA docs?

![image](https://github.com/user-attachments/assets/f832f230-c20d-487b-b3d6-ec00d6e1e956)

![image](https://github.com/user-attachments/assets/3f6008cb-72b8-42b4-ac3e-9993e132b4b5)

Reply:
Yeah, I changed it because the `JA` and `JLA` shift the upper 4 bits more:

![ja](https://github.com/user-attachments/assets/2bf2d941-778a-4852-94ea-1bf33f1ef83c)

![jla](https://github.com/user-attachments/assets/a038353b-2268-4bfc-9a08-2c52114edc2b)

Reply:
Ah wait, you mean the `-` at the beginning of the line? Yeah, this is a copy paste mistake :D

Reply:
@imbillow I am sorry. Now I got it. I didn't read properly. Of course
```7'b0000000 == 7'b0```